cmd/daemon/start.go

```diff
@@ -7,6 +7,7 @@ import (
 	"path/filepath"
 	"time"
 
+	flock "github.com/gofrs/flock"
```

Comment:
We try to avoid adding any new dependency to the project. 
Any strong reason to add it?

Reply:
Flock handles compatibility issues across different systems and provides a unified API.

Reply:
I think instead of adding a new dependency, you can use sync.RWMutex, which gives us a write locker based on the file system mechanism. For a file, you could only have one writer, and I believe this will solve the issue.

Reply:
If I understood correctly, `flock` creates a file in read-only mode and locks it [exclusively](https://github.com/gofrs/flock/blob/6f010d1acea74a32f2f2066bfe324c08bbee30e3/flock.go#L128C52-L128C56). I believe this dependency is not too heavy, so we can include it. Good find, @jkcs!